### PR TITLE
2FA security page

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR adds the 2FA security page, on new route `account/security/sudo`. I integrated the basic front elements for this page, mainly the two forms, and a new api handler `send-two-fa-validation-code` which sends the code to the chosen identity between the user primary ones, which seems to work out of the box for resending the code too. 

I modified our DynamoDB user to add new key `sudo_event_ids`, that is an array of `event_ids` objects containing the `event_id` and its expiration time. They are added to the user in `send-two-fa-validation-code` handler. 

This is still a work in progress, many things are left to do, as for the second form (the one for typing the received code), I would rework it and add its logic in another PR, or as for the mobile version. 

For now, since the page isn't working fully, I send a 404 when trying to access it for production. 

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-fd0x2p Integration of "lock screen"](https://app.clickup.com/t/fd0x2p)

File: src/pages/api/auth-connect/send-two-fa-validation-code.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      }
+
+      const identity = {
+        type: getIdentityType(identityInput.type),
+        value: identityInput.value,
+      };
+
+      return await sendTwoFAVerificationCode(config.managementCredentials, {
+        callbackUrl,
+        identity,
```
Comment (Fuzznimp) [inline src/pages/api/auth-connect/send-two-fa-validation-code.ts:76]: `callbackUrl` can be undefined, so I would do 
```suggestion
      return await sendTwoFAVerificationCode(config.managementCredentials, {
        callbackUrl: callbackUrl || "/",
        identity,
```

Maybe we could improve this by typing as possibily `undefined` somewhere, but this is not the scope of this PR.

File: src/pages/account/security/sudo.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      //   cookieName: "user-cookie",
+      //   isCookieSealed: true,
+      //   cookieSalt: config.cookieSalt,
+      // });
+
+      // if (userCookie) {
+      //   const primaryIdentities = await getIdentities(
+      //     config.managementCredentials,
+      //     userCookie.sub,
+      //   )
```
Comment (Fuzznimp) [inline src/pages/account/security/sudo.tsx:69]: I know this is still WIP, but would be cool to have
```ts
if(!userCookie) {
  //
}
```

File: src/components/pages/two-fa/two-fa.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        <p>You need double factor authentication to access this page</p>
+      </SecurityMessage>
+      <Separator />
+      <SendTwoFACodeForm
+        primaryIdentities={primaryIdentities}
+        isCodeSent={isCodeSent}
+        setIsCodeSent={setIsCodeSent}
+      />
+
+      {isCodeSent ? (
```
Comment (Fuzznimp) [inline src/components/pages/two-fa/two-fa.tsx:28]:
```suggestion
      />
      {isCodeSent ? (
```

File: src/pages/account/security/sudo.tsx (hunk 2 above)
Comment (Oluane) [reply src/pages/account/security/sudo.tsx:69]: Noted 👍

File: src/components/pages/two-fa/two-fa.tsx (hunk 3 above)
Comment (Oluane) [reply src/components/pages/two-fa/two-fa.tsx:28]: 🤖

File: src/pages/api/auth-connect/send-two-fa-validation-code.ts (hunk 1 above)
Comment (Oluane) [reply src/pages/api/auth-connect/send-two-fa-validation-code.ts:76]: We would need to do it for other flows too indeed.

File: src/pages/account/security/sudo.tsx (hunk 2 above)
Comment (Fuzznimp) [reply src/pages/account/security/sudo.tsx:69]: On this topic, everything is broken atm regarding this. I'm working on a fix ;).

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**